### PR TITLE
bug 1552159: fix code sample when no rendered HTML

### PR DESCRIPTION
**Pull request by escattone**
This fixes the bug described in https://bugzilla.mozilla.org/show_bug.cgi?id=1552159. The conditions for the bug are the following:
- `ENABLE_RESTRICTIONS_BY_HOST` is True (this is always true for stage and production, but false by default for local development)
- a request is made for a document's code sample using one of the attachment domains -- for production, these are either `mdn.mozillademos.org` or `demos-origin.mdn.mozit.cloud` 
- the document does not yet have any rendered HTML (its `rendered_html` attribute is falsy)

Under these conditions, an attempt is made to render the document, but fails when `document.get_absolute_url()` is called within `kuma.wiki.kumascript.get`, since the `wiki.document` endpoint is not defined for the attachments domains (for those domains, the root urlconf is set to `kuma.urls_untrusted`, which does not include that endpoint).

The fix is to explicitly specify the `urlconf` as `kuma.urls` when calling `reverse` within the `get_absolute_url` method.

I also modified an existing test so that it exposes the bug, preventing regressions.

### CODE REVIEW

File: kuma/wiki/tests/test_views_code.py [hunk 1]
```diff
@@ -87,18 +87,16 @@ def test_code_sample_host_allowed(code_sample_doc, settings, client):
...
-    # host restrictions.
-    code_sample_doc.get_rendered()
-
     url = reverse('wiki.code_sample',
                   args=[code_sample_doc.slug, 'sample1'])
     host = 'sampleserver'
     settings.ALLOWED_HOSTS.append(host)
     settings.ATTACHMENT_HOST = host
     settings.ENABLE_RESTRICTIONS_BY_HOST = True
+    constance_config.KUMASCRIPT_TIMEOUT = 1
```
Comment (peterbe) [inline kuma/wiki/tests/test_views_code.py]: Why this change? Is that to deliberately cause a KS timeout?

Comment (escattone) [reply kuma/wiki/tests/test_views_code.py]: Without this change, kumascript rendering is skipped (see https://github.com/mozilla/kuma/blob/aaabab98c62ebde4b4088055967568b4d9971b86/kuma/wiki/models.py#L572), so the call to `kuma.wiki.kumascript.get`, and eventually `document.get_absolute_url()`, is never made, and the problem is never exposed.

**Comment by escattone** [comment]
Thanks for the review @peterbe.

> This trick of switching urlconf, I thought that was something exclusive to the beta vs non-beta. Is it used for attachments too (unrelated to beta)?

Yes, it's used for the attachments as well.

File: kuma/wiki/tests/test_views_code.py (hunk 1 above)
Comment (peterbe) [reply kuma/wiki/tests/test_views_code.py]: It's so odd looking so could you please explain what you just explained to me in a code comment? Or do you think that's dumb because it's something everyone knows?

Comment (escattone) [reply kuma/wiki/tests/test_views_code.py]: I agree, and I've already added a comment, as you suggested, thanks!
